transcom/mymove
# Int b 19300

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19300)

## Summary

This branch contains 3 big changes:

1. The "Haul Price" item in the "Incentives" section of the final PPM Closeout Review screen will now update to read either "Linehaul Price" or "Shorthaul Price" depending on whether or not the move happens within the same Zip3.
2. This branch also fixed a bug in the pricing where items were always using the *linehaul* pricer for things like gross incentive, domestic area price, etc., even if a shipment would be classified as a shorthaul.
3. "Full Pack/Unpack Charge" has now been separated in to separate "Packing" and "Unpacking" charges.

Other minor changes:
1. "(Short/Line)haul Fuel Surcharge" has been changed to "(Short/Line)haul Fuel Rate Adjustment"

### How to test

1. Create a move that goes to any two zip codes with the same first 3 digits. Where you would normally check "use my new duty location", you can instead put in an address with the same Zip3 (e.g. 90210 -> 90245).
4. Go through the normal process to get to PPM closeout (sign, approve as SC, upload PPM documents).
5. As an SC, access the move through the closeout queue, go to the last "Confirm" screen and verify that you see "shorthaul price", "Shorthaul Fuel Rate Adjustment", and the separate "Packing Charge" and "Unpacking Charge" fields  as shown here:
![Screenshot 2024-05-23 at 2 15 09 PM](https://github.com/transcom/mymove/assets/146959895/4c60d69e-7f68-4ee5-bb41-f1a434c84472)


6. If you want to test that linehaul price will show up, find the `actual_pickup_postal_code` and `actual_destination_postal_code` columns for your move in the `ppm_shipments` table. Edit the destination code to "30183".
![Screenshot 2024-05-22 at 5 02 47 PM](https://github.com/transcom/mymove/assets/146959895/9c32142b-8a29-4a4e-bf9c-7bd408e6d1ab)

7. Refresh and continue back to the "Confirm" screen, you should now see that "Linehaul Price" and "Linehaul Fuel Rate Adjustment" shows up instead, and you should notice a substantial difference in price.

![Screenshot 2024-05-22 at 5 04 21 PM](https://github.com/transcom/mymove/assets/146959895/7aa08c0d-bfb8-4d9d-96ae-0aca5a1c3bd4)
